docs/USAGE_GUIDE.md

```diff
@@ -0,0 +1,529 @@
+</dependency>
+``` 
+
+### Create Modules
+Create two new maven modules in your project.
+
+* `environment`
+* `core`
+
+The environment module's POM should contain Scaffold's environment dependency.
+```xml
+<dependency>
+    <groupId>io.github.kgress.scaffold</groupId>
+    <artifactId>environment</artifactId>
+</dependency>
+```
+
+The core module's POM should contain Scaffold's framework dependency.
+```xml
+<dependency>
+    <groupId>io.github.kgress.scaffold</groupId>
+    <artifactId>framework</artifactId>
+</dependency>
+```
+
+### Create The Scaffold Classes
+The following are files that should be created in your project.
+
+#### Spring Configuration
+A spring configuration file is necessary for handling component scanning and the usage of spring profiles when running testing. Component scanning is necessary for dependency injection and spring profiles are used for setting
+`DesiredCapabilities` for testing. More detail regarding spring profiles are included in a later section.
+
+Create a new file under the environment module adhering to the naming convention standardization hierarchy set when creating your new project.
+
+E.G: `environment > src > main > java > your > groupID > environment > YourProjectConfiguration.java`
+
+The contents of this file should include:
+```java
+@Configuration
+@ComponentScan(value = "your.group.id")
+@PropertySource("classpath:/application.properties")
+public class YourProjectConfig {
+}
+```  
+
+#### Page Objects
+Scaffold implements the paradigm of a page object. Page objects are simple representations of web pages as a Java Object. In order to create a representation of a web page, you'll need to create a new class that contains properties
+of Scaffold's strongly typed elements. Page Objects should always extend off of the class `BasePage`, as it provides common functionality to 
+Page Objects such as `verifyIsOnPage()`.
+
+##### Strongly Typed Elements
+Scaffold makes available particular types of WebElements for you to use, which narrow the scope of methods available for any given WebElement, and keep the user focused on the actions they should be performing on these elements.
+For example, a user cannot `sendKeys()` to a Button or a Link. The `ButtonWebElement` and `LinkWebElement` objects take that into account and don't expose those methods.
+
+Another advantage of these elements is that they manage all interaction with the WebDriver internally. Most frameworks require the test write/page object maintainer to use the WebDriver to perform all their actions,
+which requires a lot of Selenium knowledge, and can also lead to race conditions, thread-safety issues, and the exposure of unnecessary complexity to the testers.
+
+In the millions of test cases run by Scaffold over the years, there has never been a reported occurrence of a StaleElementReferenceException. This is possible because the framework manages the WebDriver and the WebElements
+internally, in a thread-safe and careful manner.
```

Comment:
Could you add in the reason how stale element exceptions are not possible? I do see that they're not which is slick (when using scaffold in it's intended way).

Like linking to the code where this logic is and pointing out how any time the element is actually used, a fresh lookup is done.

Would also be good to list things that scaffold does not handle cause it becomes all native selenium stuff (but can still be avoided):

- Using an actual WebElement as the basis for any findElement(s) calls
- Re-using (i.e. not using a wrapper method to lookup lists) results of a scaffoldElement.findElement(s) call
- Using any direct WebElement instantiation instead of scaffold strong typed (i.e. having a `WebElement thing = new WebElement()` call somewhere since this is straight up native selenium)
- Creating a scaffold element using the WebElement constructor (can add more reason why this causes the same thing)

I think having this explicitly here will be super powerful when redirecting people away from doing things a certain way or when they start to use native selenium calls.

Reply:
I updated this part of the documentation to the following:

```
In the millions of test cases run by Scaffold over the years, there has never been a reported occurrence of a `StaleElementReferenceException`. This is possible because the framework manages the WebDriver and the WebElements
internally, in a thread-safe and careful manner. It circumvents the possibility of a `StaleElementReferenceException` by always re-finding the element whenever an interaction occurs on it. This behavior is only possible when using Scaffold strongly typed elements, and not the raw Selenium element.
It is strongly recommended to always Scaffold strongly typed elements in order to benefit from all Scaffold feature. Usage of raw Selenium objects will result in unexpected behavior. 
```

I have pretty heavy documentation on the constructors, too, including error logging.